### PR TITLE
Hashmap Serializer

### PR DESCRIPTION
**Pull request by TheMkrage**
Implemented a Hashmap Serializer to solve the issue with the view deleting the client’s covnoContext.

### CODE REVIEW

File: src/codeu/chat/client/core/UserContext.java [hunk 1]
```diff
@@ -97,8 +97,6 @@ public User removeConvoInterest(String title) {
 	public ConversationContext start(String name) {
 		final ConversationHeader conversation = controller.newConversation(
 				name, user.id);
-		// sets creator's id to creator in hashmap
-		conversation.setAccessOf(user.id, UserType.CREATOR);
```
Comment (KevinWorkman) [inline src/codeu/chat/client/core/UserContext.java:101]: Out of curiosity, why was this deleted? Did we never need it in the first place?

File: src/codeu/chat/server/Controller.java [hunk 2]
```diff
@@ -251,50 +250,53 @@ public Time statusUpdate() {
...
-      else
-      {
-        return null;
-      }
-
-    }
-
-    return null;
-  }
+	public Uuid changeAccessControl(Uuid userID, Uuid convo, String username,
```
Comment (KevinWorkman) [inline src/codeu/chat/server/Controller.java:263]: Side note: auto-formatting whole files makes it a little harder to focus on exactly what changed. Not really your fault, but maybe we should have come up with a better way to handle formatting.

File: src/codeu/chat/util/Serializers.java [hunk 3]
```diff
@@ -150,6 +151,31 @@ public void write(OutputStream out, Collection<T> value) throws IOException {
       }
     };
   }
+  
+  public static <K, V> Serializer<HashMap<K, V>> hashmap(final Serializer<K> keySerializer, final Serializer<V> valueSerializer) {
```
Comment (KevinWorkman) [inline src/codeu/chat/util/Serializers.java:155]: comments comments comments

**Review by KevinWorkman — COMMENTED** [review]
Looks good. Would really like to see some tests if at all possible.

File: src/codeu/chat/client/core/UserContext.java (hunk 1 above)
Comment (TheMkrage) [reply src/codeu/chat/client/core/UserContext.java:101]: Correct, it wasn't needed since owner is passed as a parameter when creating a convo

File: src/codeu/chat/server/Controller.java (hunk 2 above)
Comment (TheMkrage) [reply src/codeu/chat/server/Controller.java:263]: Debating doing a PR where every file is just autoformatted :P

File: src/codeu/chat/util/Serializers.java (hunk 3 above)
Comment (TheMkrage) [reply src/codeu/chat/util/Serializers.java:155]: Done. Can you make any suggestions on how I would write tests for this? I don't really undersatand how to simulate the server environment since the streams come from the server wherever this serializer is used. Thanks.
